src/main/java/org/opensearch/geospatial/index/mapper/xyshape/XYShapeSupportVisitor.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.index.mapper.xyshape;
+
+import static org.opensearch.geometry.ShapeType.CIRCLE;
+import static org.opensearch.geometry.ShapeType.LINEARRING;
+
+import java.util.Locale;
+import java.util.Objects;
+
+import org.opensearch.geometry.Circle;
+import org.opensearch.geometry.Geometry;
+import org.opensearch.geometry.GeometryCollection;
+import org.opensearch.geometry.GeometryVisitor;
+import org.opensearch.geometry.Line;
+import org.opensearch.geometry.LinearRing;
+import org.opensearch.geometry.MultiLine;
+import org.opensearch.geometry.MultiPoint;
+import org.opensearch.geometry.MultiPolygon;
+import org.opensearch.geometry.Point;
+import org.opensearch.geometry.Polygon;
+import org.opensearch.geometry.Rectangle;
+
+// Visitor to check whether shape is supported for indexing or not
+public final class XYShapeSupportVisitor implements GeometryVisitor<Geometry, RuntimeException> {
```

Comment:
what is the purpose of these visitors? For GeoShape it made sense as we were normalizing the lat lon present in the Shape.

Also, if we are only doing null checks cannot we do this with 1 single function?

Reply:
Super class will call prepare indexing method to determine whether to support particular Geometry or not. Here, we are doing null check for supported geometry and raising unSupportedException for other objects

Reply:
Point p = null;
visitor.visit(p)